src/block_ffm.rs

```diff
@@ -542,16 +554,53 @@ impl<L: OptimizerTrait + 'static> BlockTrait for BlockFFM<L> {
                             is_first_feature = false;
                             contra_fields_copied = true;
                             // Copy only once, skip other copying as the data for all features of that contra_index is already calculated
-                            contra_fields
-                                .get_unchecked_mut(offset..offset + field_embedding_len_as_usize)
-                                .copy_from_slice(
-                                    cached_contra_fields.get_unchecked(
-                                        offset..offset + field_embedding_len_as_usize,
-                                    ),
-                                );
+                            ptr::copy_nonoverlapping(
+                                cached_contra_fields.as_ptr().add(offset),
+                                contra_fields.as_mut_ptr().add(offset),
+                                field_embedding_len_as_usize,
+                            );
                         } else if !contra_fields_copied {
                             contra_fields_copied = true;
-                            for z in 0..field_embedding_len_as_usize {
+
+                            const LANES: usize = STEP * 4;
+                            let field_embedding_len_end = field_embedding_len_as_usize
+                                - (field_embedding_len_as_usize % LANES);
```

Comment:
if lanes is 2**sth (it seems to be) you could further reduce the amount of instructions by & (LANES - 1)-ing it

Reply:
I do a step over everything as I need to move the pointer forward

Reply:
Not sure I follow -> the suggestion was to get rid of modulo operator which on paper costs more than binary and-ing it appropriately